From aabfeb71cd7a7593a15a6591ac10df3d37393d2f Mon Sep 17 00:00:00 2001
From: blobfish <blobfish@gmx.com>
Date: Sat, 23 Apr 2022 10:59:52 -0400
Subject: [PATCH 10/18] BRepFill_OffsetWire: segfault. See following:

    bfb# 000004
    freecad thread: https://forum.freecadweb.org/viewtopic.php?f=10&t=67881

    throw Standard_OutOfRange out of MAT_TList::Current() before illegal access.
---
 src/MAT/MAT_TList.gxx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/MAT/MAT_TList.gxx b/src/MAT/MAT_TList.gxx
index 15d1f03d3f..87f2cabb3c 100644
--- a/src/MAT/MAT_TList.gxx
+++ b/src/MAT/MAT_TList.gxx
@@ -14,6 +14,8 @@
 // Alternatively, this file may be used under the terms of Open CASCADE
 // commercial license or contractual agreement.
 
+#include <Standard_OutOfRange.hxx>
+
 //=======================================================================
 //function : MAT_TList
 //purpose  : 
@@ -107,6 +109,8 @@ Standard_Boolean MAT_TList::More() const
 
 Item MAT_TList::Current() const
 {
+  if (thecurrentnode.IsNull())
+    throw Standard_OutOfRange();
   return thecurrentnode->GetItem();
 }
 
-- 
2.45.0

